src/ldp/auxiliary/RdfValidator.ts

```diff
@@ -23,23 +22,22 @@ export class RdfValidator extends Validator {
     if (representation.metadata.contentType === INTERNAL_QUADS) {
       return;
     }
-
-    // eslint-disable-next-line unicorn/expiring-todo-comments
-    // TODO: Everything below should be part of a utility cloneRepresentation function.
-
     const identifier = { path: representation.metadata.identifier.value };
-
-    // Read data in memory first so it does not get lost
-    const data = await arrayifyStream(representation.data);
     const preferences = { type: { [INTERNAL_QUADS]: 1 }};
-
-    // Creating new representation since converter might edit metadata
-    const tempRepresentation = new BasicRepresentation(data, identifier, representation.metadata.contentType);
-    const result = await this.converter.handleSafe({ identifier, representation: tempRepresentation, preferences });
+    let result;
+    try {
+      // Creating new representation since converter might edit metadata
+      const tempRepresentation = await cloneRepresentation(representation);
+      result = await this.converter.handleSafe({
+        identifier,
+        representation: tempRepresentation,
+        preferences,
+      });
+    } catch {
+      representation.data.destroy();
+      throw new Error('bad data!');
```

Comment:
```suggestion
    } catch (error: unknown) {
      representation.data.destroy();
      throw error;
```
We still want to throw the same error.